File: Data Structures & Algorithms/two-integer-sum/submission-3.cpp
class Solution {
public:
    vector<int> twoSum(vector<int>& nums, int target) {
        unordered_map<int, int> indexes;
        int n = nums.size();
        for (int i = 0; i < n; ++i) {
            if (indexes.contains(target - nums[i])) {
                return {indexes[target - nums[i]], i};
            }
            indexes[nums[i]] = i;
        }
        return {};
    }
};

File: Data Structures & Algorithms/two-integer-sum-ii/submission-3.cs
public class Solution {
    public int[] TwoSum(int[] numbers, int target) {
        int l = 0;
        int r = numbers.Count() - 1;

        while (l < r) {
            int total = numbers[l] + numbers[r];

            if (total == target) {
                return [1 + l, 1 + r];
            }
            else if (total > target) {
                --r;
            }
            else {
                ++l;
            }
        }
        return [-1, -1];
    }
}
